evo_researcher/benchmark/benchmark.py

```diff
@@ -281,6 +289,50 @@ def get_markets_summary(self) -> t.Dict[str, t.List[str]]:
             ]
         markets_summary[f"reference p_yes"] = [m.p_yes for m in self.markets]
         return markets_summary
+    
+    def calculate_expected_value(self, prediction: Prediction, market: Market):
+        if not prediction.is_answered:
+            return None
+
+        # TODO: Add support for different bet sizes and calculate shares based on the market's odds.
+        bet_units = 10  # Assuming the agent always bet 10 units per market.
+        receive_shares = 20  # Because we assume markets trades at 50/50.
+        buy_yes_threshold = 0.5  # If the agent's prediction is > 50% it should buy "yes", otherwise "no".
+
+        yes_shares = receive_shares if prediction.outcome_prediction.p_yes > buy_yes_threshold else 0
+        no_shares = receive_shares if prediction.outcome_prediction.p_yes <= buy_yes_threshold else 0
+        
+        expected_value = (
+            yes_shares * market.p_yes  
+            + no_shares * (1 - market.p_yes)
+            - bet_units
+        )
+
+        return expected_value
+
+    def compute_expected_values_summary(self):
+        overall_summary = defaultdict(list)
+
+        for agent in self.registered_agents:
+            expected_values = []
+
+            for market in self.markets:
+                if (prediction := self.get_prediction(agent.agent_name, market.question)).is_answered:
+                    expected_values.append(self.calculate_expected_value(prediction, market))
+
+            overall_summary["Agent"].append(agent.agent_name)
+            overall_summary["Mean expected value"].append(np.mean(expected_values))
+            overall_summary["Total expected value"].append(np.sum(expected_values))
```

Comment:
I thought it wouldn't hurt to have more of them. Having `sum` is nice to know some final stat, but having mean could be good to compare benchmarks with a different number of markets, too.

Maybe add a median as well? 🤔

Reply:
Oh good point, because agents can choose to only answer a subset of the markets, so an agent might have a lower mean but higher total expected value. My bad